src/SideSheet/index.js

```diff
@@ -0,0 +1,188 @@
+    }
+  };
+
+  const closeTransition = () => {
+    setTransition(false);
+    setTimeout(() => setOpen(false), 500);
+  };
+
+  useEffect(() => {
+    setSideSheetPosition({
+      transition: `transform .4s cubic-bezier(.3,0,0,1)`,
+      transform: transition
+        ? `translateX(calc(100vw - ${width}px - 20px))`
+        : `translateX(100vw)`,
+      top: 0,
+      bottom: 0,
+      width: width,
+      height: "calc(100% - 20px)",
+    });
+  }, [open, transition]);
+
+  useEffect(() => {
+    document.addEventListener("mousedown", onOutSideClick);
+
+    return () => {
+      document.removeEventListener("mousedown", onOutSideClick);
+    };
+  }, []);
+
+  return (
+    <div
+      ref={sideSheet}
+      className={classNames("relative inline-block text-left", className)}
+    >
+      {cloneElement(children, {
+        onClick: (e) => {
+          e.stopPropagation();
+          setTransition(false);
+          if (!open) {
+            setOpen(true);
+            setTimeout(() => setTransition(true), 0);
+          }
+        },
+      })}
+      {open && (
+        <SideSheetContent id="sidesheet">
+          <div
+            className={classNames(
+              "fixed z-50 inset-0 opacity-25 duration-300 delay-200 transition",
+              {
+                "bg-gray-300": transition,
+                "bg-transparent": !transition,
+              }
+            )}
+          />
+          <div
+            ref={portal}
+            style={{
+              ...sideSheetPosition,
+            }}
+            className={
+              "fixed z-50 min-w-0 bg-white duration-300 delay-200 h-full flex flex-col shadow-xl m-2 rounded"
+            }
+          >
+            <Box
+              flex
+              alignItems="center"
+              className="relative border-b border-gray-200 rounded rounded-b-none"
+            >
+              <div className="text-gray-600 flex-auto truncate leading-6">
+                {title}
+              </div>
+              <Box
+                flex
+                backgroundColor="transparent"
+                className="cursor-pointer"
+                noPadding
+              >
+                <div onClick={() => closeTransition()}>
+                  <XIcon />
+                </div>
+              </Box>
+            </Box>
+            <Box
+              flex
+              flexDirection="col"
+              className="sidesheet-content relative overflow-y-auto flex-1"
+            >
+              <Box noPadding flex flexDirection="col" className="h-full">
+                {content}
+              </Box>
+            </Box>
+            <Box
+              flex
+              justifyContent="center"
+              alignItems="center"
+              className="sidesheet-action h-16 border-t border-gray-200 relative flex-initial rounded rounded-t-none"
+            >
+              <Button onClick={() => onButtonClick()} label={buttonText} />
+            </Box>
```

Comment:
I think we should make the field open by removing the Button component and having a conditional to show or hide the whole block based on the action prop existing

Reply:
@esoubihe sorry, I didn't understand exactly how it should works.

Reply:
If `action` is set, show 

```javascript
           <Box
              flex
              justifyContent="center"
              alignItems="center"
              className="sidesheet-action h-16 border-t border-gray-200 relative flex-initial rounded rounded-t-none"
            >
              ${action}
            </Box>
```
else return nothign

Reply:
ahhh that is ok. Only show the button if the action params exists, makes sense.

Reply:
done!